tasks/python-venv_installation.yml

```
@@ -0,0 +1,104 @@
+---
+
+- name: python venv | check if the {{ env_name }} dir exists
+  stat:
+    path: "{{ virtualenv_basedir }}/{{ env_name }}"
+  register: venv_dir
+
+- name: python venv | register existing venv python version
+  become: true
+  ansible.builtin.command: "{{ virtualenv_basedir }}/{{ env_name }}/bin/python --version"
+  register: host_venv_vers
+  when: venv_dir.stat.exists
+  changed_when: false
+
+- name: python venv | Set a host_venv_version
+  ansible.builtin.set_fact:
+    host_venv_version: "{{ host_venv_vers.stdout | ansible.builtin.regex_replace ('Python ''')| regex_search('^\\d+\\.\\d+') | string}}"
+  when: venv_dir.stat.exists
+
+- name: python venv | Copy venv to a folder whose name includes the python version
+  become: true
+  ansible.builtin.copy:
+    remote_src: True
+    src: "{{ virtualenv_basedir }}/{{ env_name }}/"
+    dest: "{{ virtualenv_basedir }}/venv-{{ host_venv_version }}/"
+  when: venv_dir.stat.exists and not venv_dir.stat.islnk
+
+- name: python venv | Display Warning Message
+  ansible.builtin.debug:
+      msg: "The venv is upgraded to use a symlink, the service ({{ service_name }}) will be stopped before the changes and restarted afterwards."
+  when:
+    - venv_dir.stat.exists
+    - not venv_dir.stat.islnk
+    - python_version == host_venv_version
+    - service_name  | length > 0
+
+- name: python venv | stop the server
```

```suggestion
- name: python venv | stop the service
```

This can be also ``omero-web`` service.